R/global_search.R

```diff
@@ -0,0 +1,170 @@
+  }
+
+
+  tryCatch({
+    res <- .request('POST', path="v2/search", body=body, env=env)
+    res <- formatSolveBioQueryResponse(res)
+
+    return(res)
+  }, error = function(e) {
+    cat(sprintf("Query failed: %s\n", e$message))
+  })
+}
+
+
+#' GlobalSearch.results
+#'
+#' Performs a Global Search based on provided filters, entities, queries, and returns an R data frame containing results from API response.
+#' Returns a single page of results otherwise (default).
+#'
+#' @param paginate When set to TRUE, retrieves all records (memory permitting).
+#' @param env (optional) Custom client environment.
+#' @param use_field_titles (optional) Use field title instead of field name for query.
+#' @param ... (optional) Additional query parameters (e.g. filters, entities, query, limit, offset).
+#'
+#' @examples \dontrun{
+#' # No filters applied
+#' GlobalSearch.results()
+#'
+#' #Global Beacon search
+#' GlobalSearch.results(entities = '[["gene","BRCA2"]]')
+#'
+#  #Type filter (only vaults)
+#' GlobalSearch.results(filters = '[{"and":[["type__in",["vault"]]]}]')
+#'
+#' # Advanced search
+#' GlobalSearch.results(query = "fuji")
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+GlobalSearch.results <- function(paginate=FALSE, use_field_titles=TRUE, env = solvebio:::.solveEnv, ...) {
+  params <- list(...)
+  params$env <- env
+
+  # Retrieve the first page of results
+  response <- do.call(GlobalSearch.search, params)
+  df <- response$result
+  offset <- response$offset
+
+  if (response$total > 100000 && isTRUE(paginate)) {
+    warning(paste("This query will retrieve ", response$total, " records, which may take some time..."), call. = FALSE)
+  }
+
+  # Continue to make requests for data if pagination is enabled and there are more records
+  while (isTRUE(paginate) && !is.null(offset)) {
+    params$offset <- offset
+    response <- do.call(GlobalSearch.search, params)
+    df_page <- response$results
+    df <- dplyr::bind_rows(df, df_page)
+    offset <- response$offset
+  }
+
+  if (!isTRUE(paginate) && !is.null(offset)) {
+    warning(paste("This call returned only the first page of records. To retrieve more pages automatically,",
+                  "please set paginate=TRUE when calling Dataset.query().", call. = FALSE))
+  }
+
+
+  return(df)
+}
+
+
+#' GlobalSearch.subjects
+#'
+#' Performs a Global Search based on provided filters, entities, queries, and returns an R data frame containing subjects from API response.
+#' @param env (optional) Custom client environment.
+#' @param ... (optional) Additional query parameters (e.g. filters, entities, query, limit, offset).
+#'
+#' @examples \dontrun{
+#' GlobalSearch.subjects(entities = '[["gene","BRCA2"]]')
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+GlobalSearch.subjects <- function(env = solvebio:::.solveEnv, ...) {
+  params <- list(...)
+  params$env <- env
+
+  # Retrieve the first page of subject
```

Comment:
There is generally just 1 page of results for this, so we can remove this comment